fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -158,7 +163,8 @@ class ApplyMetadataMutationsImpl {
 
 private:
 	void writeMutation(const MutationRef& m) {
-		if (forResolver || !SERVER_KNOBS->ENABLE_TLOG_ENCRYPTION) {
+		if (forResolver ||
+		    !isEncryptionEnabled(EncryptOperationType::TLOG_ENCRYPTION, dbInfo->get().client.isEncryptionEnabled)) {
```

Comment:
Do we want to hide the details about `flag used in ClientDBInfo` and just pass instance of `dbInfo` instead? For instance: we know the `ClientDBInfo` flag name/type may change in near future, abstracting such details will make switching to newer format with much ease. Thoughts?